qoqo-iqm/src/backend.rs

```diff
@@ -344,15 +348,18 @@ impl BackendWrapper {
     ///     RuntimeError: Something went wrong when submitting the job to the backend.
     pub fn submit_circuit_batch(&self, circuits: Vec<PyObject>) -> PyResult<String> {
         let mut circuit_batch: Vec<Circuit> = Vec::new();
-        for circuit in circuits.into_iter() {
-            let tmp_circuit = CircuitWrapper::from_pyany(circuit).map_err(|err| {
-                PyTypeError::new_err(format!(
-                    "`circuits` argument is not a list of qoqo Circuits: {}",
-                    err
-                ))
-            })?;
-            circuit_batch.push(tmp_circuit)
-        }
+        Python::with_gil(|py| -> PyResult<()> {
```

Comment:
Why do we need to hold the GIL here? Is this needed by pyo3 0.21?

Reply:
With the updated to pyo3 0.21 `CircuitWrapper::from_pyany` now takes a `&Bound<PyAny>` and the `bind` function to convert circuit that is a `Py<PyAny>` to a  `&Bound<PyAny>` needs the py context

Reply:
I see. Maybe we can change the input to this function from `Vec<PyObject>` to `Vec<&Bound<PyAny>>` , so we would not need to call `bind` at all? Not sure if this would work